snuba/query/validation/validators.py

```diff
@@ -55,6 +71,38 @@ def validate(self, query: Query, alias: Optional[str] = None) -> None:
             )
 
 
+class EntityContainsColumnsValidator(QueryValidator):
+    """
+    Ensures that all columns in the query actually exist in the entity.
+    """
+
+    def __init__(
+        self, entity_data_model: ColumnSet, validation_mode: ColumnValidationMode
+    ) -> None:
+        self.validation_mode = validation_mode
+        self.entity_data_model = entity_data_model
+
+    def validate(self, query: Query, alias: Optional[str] = None) -> None:
+        if self.validation_mode == ColumnValidationMode.DO_NOTHING:
+            return
+
+        query_columns = query.get_all_ast_referenced_columns()
+
+        missing = set()
+        for column in query_columns:
+            if column.table_name == alias and column.column_name not in {
+                c.name for c in self.entity_data_model.columns
+            }:
+                missing.add(column.column_name)
+
+        if missing:
+            error_message = f"query column(s) {', '.join(missing)} do not exist"
+            if self.validation_mode == ColumnValidationMode.ERROR:
+                raise InvalidQueryException(error_message)
+            elif self.validation_mode == ColumnValidationMode.WARN:
+                logger.warning(error_message)
```

Comment:
I'm pretty sure you have to have `exc_info=True` in the logging call to get them sent to Sentry.